src/channel_state.ts

```diff
@@ -207,6 +221,7 @@ export class ChannelState<
       ...message,
       __html: message.html,
       // parse the date..
+      pinned_at: message.pinned_at ? new Date(message.pinned_at) : new Date(),
```

Comment:
shouldn't this be null if the message is not pinned?

Reply:
It will become `Date | null` which is impossible to reuse in `_addToMessageList`

Reply:
Nevermind, it is broken. Fixing...